com.unity.cinemachine/Samples~/Shared Assets/UI/SamplesDynamicUI.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UIElements;
+
+namespace Unity.Cinemachine.Samples
+{
+    /// <summary>
+    /// Runtime UI script that can create toggles and buttons.
+    /// </summary>
+    [RequireComponent(typeof(UIDocument))]
+    public class SamplesDynamicUI : MonoBehaviour
+    {
+        [Serializable]
+        public class Item
```

Comment:
Something like this will make a nicer inspector:
```        [Serializable]
        public class Item
        {
            [Tooltip("Text displayed on button")]
            public string Name = "Text";

            [Serializable]
            public class ToggleSettings
            {
                public bool Enabled;

                [Tooltip("Initial value of the toggle")]
                public bool Value;

                [Tooltip("Event sent when toggle button is clicked")]
                public UnityEvent<bool> OnValueChanged = new();
            }

            [Tooltip("Set this to true to create a toggle button")]
            [FoldoutWithEnabledButton]
            public ToggleSettings IsToggle = new();

            [Tooltip("Event sent when button is clicked")]
            public UnityEvent OnClick = new();
        }